server/machine_interfaces/haas.py

```diff
@@ -0,0 +1,192 @@
+# Luke Minton, MDE Fall 2021, VT ISE Learning Factory, Interface from Haas CNC machine to LF server
+# Machine currently has a serial (RS232) to USB port
+
+import serial, time, sys, zmq, pickle
+from typing import Dict, Any
+from threading import Thread
+from base import MachineInterface
+from lf_utils import retry
+
+# Haas macros for pulling data
+COOLANT_LEVEL = b"?Q600 1094\r\n"
+SPINDLE_SPEED = b"?Q600 3027\r\n"
+MACHINE_COORD_X = b"?Q600 5021\r\n"
+MACHINE_COORD_Y = b"?Q600 5022\r\n"
+MACHINE_COORD_Z = b"?Q600 5023\r\n"
+MACHINE_COORD_A = b"?Q600 5024\r\n"
+MACHINE_COORD_B = b"?Q600 5025\r\n"
+WORK_COORD_X = b"?Q600 5041\r\n"
+WORK_COORD_Y = b"?Q600 5042\r\n"
+WORK_COORD_Z = b"?Q600 5043\r\n"
+WORK_COORD_A = b"?Q600 5044\r\n"
+WORK_COORD_B = b"?Q600 5045\r\n"
```

Comment:
Put these in a config and load from config.

In general it is poor practice to mix code and constants. We want to decouple _how something is done_ from _the way it is parameterized_.